### PR TITLE
Update BalanceFetcher with updating tokens using .main queue #3125

### LINKED ISSUE #3125: Reduce background queues count
Fix rapid Realm files grow up, move from background accessing to realm to using main thread

### PR DESCRIPTION
**Pull request by vladyslav-iosdev**
Closes #3125

### CODE REVIEW

**Comment by vladyslav-iosdev** [comment]
Wait until erc1155 will done.

**Comment by vladyslav-iosdev** [comment]
@hboon u can check this pr, i suppose its need to rename it ;)

File: AlphaWallet/Core/Coordinators/WalletBalance/PrivateBalanceFetcher.swift [hunk 1]
```diff
@@ -89,29 +90,32 @@ class PrivateBalanceFetcher: PrivateBalanceFetcherType {
...
-                }
-
-                if let value = balanceValueHasChange, value {
-                    delegate.didUpdate(in: strongSelf)
-                }
-
-                group.leave()
-            }
+            let promise = refreshBalanceAction(forToken: tokenObject)
+            promises.append(promise)
```
Comment (hboon) [inline AlphaWallet/Core/Coordinators/WalletBalance/PrivateBalanceFetcher.swift:111]: We can use a `map()`?

**Review by hboon — APPROVED** [review]
Looks better. Too many `DispatchGroup`s are harder to track :O

File: AlphaWallet/Core/Coordinators/WalletBalance/PrivateBalanceFetcher.swift (hunk 1 above)
Comment (vladyslav-iosdev) [reply AlphaWallet/Core/Coordinators/WalletBalance/PrivateBalanceFetcher.swift:111]: yes, but this removes them at all.

**Comment by vladyslav-iosdev** [comment]
yep, with these changes the app works a little bit slower, but the issues we got with threads should disappear, as well as crushes related to them

**Comment by vladyslav-iosdev** [comment]
but it seems like it stucks on app launch only.

**Comment by vladyslav-iosdev** [comment]
@hboon i am not pretty sure but in `TokensDataStore` we got `updateTokenUnsafe(primary Key` method, that checks whethed value should be changed or not, what do u think, do we need to do this check? it shouldn't brake any logic, its just for safety to avoid notifying that object has changed,  when its value hasn't changed

**Comment by hboon** [comment]
@vladyslav-iosdev I haven’t checked. But sounds like it’ll be fine. So you want to merge this into master? We can run with it and observe like you said

**Pull request merged**